bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/TTSService.java

```diff
@@ -72,4 +75,34 @@ public interface TTSService {
      *             {@link AudioStream}
      */
     public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFormat) throws TTSException;
+
+    /**
+     * Check if this TTS service accepts the system-wide TTS cache service.
+     *
+     * @return
+     */
+    public default boolean isCacheEnabled() {
+        return true;
+    }
+
+    /**
+     * Construct a uniquely identifying string for the request. Could be overridden by the TTS service if
+     * it uses some unique external parameter and wants to identify variability in the cache.
+     *
+     * @param text The text to convert to speech
+     * @param voice The voice to use for speech
+     * @param requestedFormat The audio format to return the results in
+     * @return A likely unique key identifying the combination of parameters and/or internal state,
+     *         as a string suitable to be part of a filename. This will be used in the cache system to store the result.
+     */
+    public default String getCacheKey(String text, Voice voice, AudioFormat requestedFormat) {
```

Comment:
OK. I see two options here : 

- "can be" extended : if TTS service are not compeled to extends the AbstractTTSService, then I have to check with instanceof and use alternative method for getting "getCacheKey" (or "isCacheEnabled" if we keep it)

- "must be" extend : no alternative method needed, but every TTS service have to be changed accordingly

WDYT ?

Reply:
You could also add an interface `CachedTTSService` which extends `TTSService`. This is probably even a better idea, because it can separate the cache methods from the `TTSService` methods. TTS services that don't want to (or can't) use the cache implement `TTSService`, others `CachedTTSService` (similar to `PersistenceService` and `ModifiablePersistenceService`). This also removes the need to programatically disable the cache.

Reply:
Done, with a proposal of separation of concerns : 
The TTSCache is now an OSGI service (with TTSLRUCacheImpl implementing it). It respects the global parameter to enable the cache or not.
VoiceManagerImpl is left totally untouched, not tied to the cache anymore.
Each TTSService can choose to use the TTSCache or not.
To do so, they can extend a helper Class (AbstractCachedTTSService, which implements the standard TTSService). The TTSCache  is injected inside and used transparently.